### ISSUE TITLE
[Bug] 5.6: ember-template-compiler cannot used Application.initializer, as it could not be a function

### ISSUE THREAD

**Issue by NullVoxPopuli** [emberjs/ember.js#20640]
### 🐞 Describe the Bug

![image](https://github.com/emberjs/ember.js/assets/199018/6ef807be-bfee-4a4f-8b14-6fbb5bfe6806)

### 😕 Actual Behavior

error

### 🤔 Expected Behavior

template-compiler loads at runitme.

(I know this'll be fixed by the new runtime compiler, but that isn't available yet, so folks needing to use runtime compilation must stay on, at newest, ember-source 5.5)

### 🌍 Environment

- Ember: - 5.6

### ➕ Additional Context

Loaded via: ember-cli-build.js (with embroider / webpack)
```js
  // Adds:
  //  - ember-template-compiler
  //  - @glimmer/syntax
  app.import('vendor/ember/ember-template-compiler.js');
```

**Comment by NullVoxPopuli** [comment]
this is resolved if `statkcEmberSource` is turned off

**Comment by boris-petrov** [comment]
So this is a problem only under Embroider? I didn't see such an issue in my case using `ember-cli`.

**Comment by NullVoxPopuli** [comment]
yeah, older `ember-cli` ships everything in node_modules, and there is no `staticEMberSource` option :sweat_smile:

**Comment by NullVoxPopuli** [comment]
Looks like this happens for tests, too:
```
Uncaught TypeError: ember__WEBPACK_IMPORTED_MODULE_6__.default.Test is undefined
    setupTestAdapter index.js:123
    start index.js:203
    js test-helper.js:13
    Webpack 7
[index.js:123:36](webpack://docs-app/src/index.js)
Uncaught Error: The tests file was not loaded. Make sure your tests index.html includes "assets/tests.js".
```

**Comment by NullVoxPopuli** [comment]
I haven't seen this in ages -- I suspect because we shipped an actually supported template compiler API (for a while now)
